### PR TITLE
Refactoring install plugin script

### PR DESCRIPTION
**Pull request by geet-h17**
## Changes Proposed

Refactored the installation script to improve functionality, efficiency, and maintainability.

## Additional Information

I have made the following changes to the script:
- Replaced inconsistent variable usage with consistent usage of the `name` variable.
- Introduced functions to encapsulate repetitive code blocks.
- Handled errors gracefully and provided informative error messages.
- Validated the availability of required dependencies.
- Used consistent indentation and formatting throughout the script.
- Separated concerns into separate functions for version detection, downloading, and installation.

## Check List

- [ x] The title of my pull request is a short description of the changes
- [ ] This PR relates to some issue: 
- [ x] I have documented the changes made (if applicable)
- [ x] I have covered the changes with unit tests

### CODE REVIEW

File: scripts/install_plugin.sh [hunk 1]
```diff
@@ -1,53 +1,87 @@
...
 [ -z "$version" ] && {
-  version="$(cat plugin.yaml | grep "version" | cut -d '"' -f 2)"
-  echo Defaulted to version: $version
+    version="$(awk -F '"' '/version/ {print $2}' plugin.yaml)"
+    echo "Defaulted to version: $version"
 }
-echo "Downloading and installing ${name} v${version} ..."
 
-url=""
+echo "Downloading andinstalling ${name} v${version} ..."
```
Comment (undera) [inline scripts/install_plugin.sh:63]: Misprint: space

File: scripts/install_plugin.sh [hunk 2]
```diff
@@ -1,53 +1,87 @@
...
+    echo "$1" >&2
+    exit 1
+}
+
+# Function to validate command availability
+validate_command() {
+    command -v "$1" >/dev/null 2>&1 || error_exit "Required command '$1' not found. Please install it."
+}
 
-[ ! -z "$HELM_DEBUG" ] && set -x
```
Comment (undera) [inline scripts/install_plugin.sh:5]: What happened to HELM_DEBUG logic? The intent there is to look at that env variable and add bash command verbosity.

**Review by undera — CHANGES_REQUESTED** [review]
Fix misprint please, and address my comment

File: scripts/install_plugin.sh (hunk 2 above)
Comment (geet-h17) [reply scripts/install_plugin.sh:5]: sorry it was just an unintentional deletion as I typed wrong number of lines to be dd-ed on vim... I have update the code now.

**Comment by undera** [comment]
Thanks for the contribution!

**Pull request merged**